src/H5CX.c

```diff
@@ -1185,6 +1194,7 @@ H5CX_set_dcpl(hid_t dcpl_id)
     H5CX_node_t **head = NULL; /* Pointer to head of API context list */
 
     FUNC_ENTER_NOAPI_NOINIT_NOERR
+    H5_API_LOCK
```

Comment:
As much as I hate the proliferation of the FUNC_ENTER/LEAVE macros, it may make sense to have a pair for functions like this that still need to acquire the lock immediately on entry and release it right before exit. Otherwise, the additional function prologue/epilogue macros become a burden and create opportunities for accidentally missing one or getting them in the wrong order.